cmd/pj-rehearse/main.go

```diff
@@ -228,29 +228,30 @@ func main() {
 	rehearsals := rehearse.ConfigureRehearsalJobs(toRehearse, prConfig.CiOperator, prNumber, loggers, o.allowVolumes, changedTemplates)
 	if len(rehearsals) == 0 {
 		logger.Info("no jobs to rehearse have been found")
-		gracefulExitWithConfigMapCleanup(true, "", cmManager)
+		return 0
 	} else if len(rehearsals) > o.rehearsalLimit {
 		jobCountFields := logrus.Fields{
 			"rehearsal-threshold": o.rehearsalLimit,
 			"rehearsal-jobs":      len(rehearsals),
 		}
 		logger.WithFields(jobCountFields).Info("Would rehearse too many jobs, will not proceed")
-		gracefulExitWithConfigMapCleanup(true, "", cmManager)
+		return 0
 	}
 
 	executor := rehearse.NewExecutor(rehearsals, prNumber, o.releaseRepoPath, jobSpec.Refs, o.dryRun, loggers, pjclient)
 	success, err := executor.ExecuteJobs()
 	if err != nil {
 		logger.WithError(err).Error("Failed to rehearse jobs")
-		gracefulExitWithConfigMapCleanup(o.noFail, rehearseFailureOutput, cmManager)
+		return gracefulExit(o.noFail, rehearseFailureOutput)
 	}
 	if !success {
 		logger.Error("Some jobs failed their rehearsal runs")
-		gracefulExitWithConfigMapCleanup(o.noFail, jobsFailureOutput, cmManager)
+		return gracefulExit(o.noFail, jobsFailureOutput)
 	}
 	logger.Info("All jobs were rehearsed successfully")
+	return 0
+}
 
-	if err := cmManager.CleanupCMTemplates(); err != nil {
-		logger.WithError(err).Error("Failed to cleanup temporary configmaps")
-	}
+func main() {
+	os.Exit(rehearseMain())
```

Comment:
If we're doing this we can just `defer` the cleanup call at the top of `rehearseMain()` and not need to do the return value stuff.

Reply:
How would we distinguish the success (`os.Exit(0)`) from failure (`os.Exit(1)`) of the whole rehearsal?